darwin/dataset/download_manager.py

```diff
@@ -94,7 +94,7 @@ def download_all_images_from_annotations(
 
     # Verify that there is not already image in the images folder
     unfiltered_files = images_path.rglob(f"*") if use_folders else images_path.glob(f"*")
-    existing_images = {image.stem: image for image in unfiltered_files if is_image_extension_allowed(image.suffix)}
+    existing_images = [image for image in unfiltered_files if is_image_extension_allowed(image.suffix)]
```

Comment:
is there a reason to drop the dictionary here? Dictionaries have constant time lookup, so the checks later like if annotation.filename in existing_images are very quick on large datasets, but over an array it has to iterate over until it finds it (or the whole array worst case when it's not)

Reply:
If just need the name rather than key:value for a dictionary, a set is the same as dictionary lookup and has the same comprehension notation just without the key:value `{x for x in list}`

Reply:
The reason I changed to list is that the way this dict is constructed results in overwriting of keys. If there is more than 1 identical `image.stem` object in the release, then that key is written twice. This isn't a problem for appending to a list since we're only recording paths w/out stems

Given this, it sounds like a set is the best option, thanks for the knowledge!